### DISCUSSION TITLE
Définition de templates pour les dossiers et noms de fichiers (autre?)

### DISCUSSION THREAD

**Discussion by PBoissier** [IPGP/autorino#1] [Ideas]
## Fil de discussion pour le choix de la syntaxe des templates pour les dossiers et noms de fichiers (autre?)

**Comment by PBoissier** [comment]
Ma proposition : utiliser le moteur de template Jinja.

Les variables (ou expressions) sont définies  dans des balises `{{ }}`

Exemple : 

```
import jinja2

# Directory template as defined in the configuration file
directory_template_string = "{{ code9 }}/{{ yyyy }}/{{ doy }}"
# Directory template as defined in the configuration file
file_name_template_string = "{{ code9 }}{{ yyyy }}{{ hh }}.t02"

# Example of station informations
code9 = "SNEG"
yyyy = "2022"
doy = "340"
hh = "00"

## Jinja magic!
# Environment initialisation
environment = jinja2.Environment()
# Directory template initialisation
directory_template = environment.from_string(directory_template_string)
# Data file name template initialisation
file_name_template = environment.from_string(file_name_template_string)
# Output tests
print("Dossier : %s" %(directory_template.render(code9=code9, yyyy=yyyy, doy=doy)))
print("Fichier : %s" %(file_name_template.render(code9=code9, yyyy=yyyy, hh=hh)))
```

**Reply by PBoissier** [reply]
Jinja est très utilisé sur les framework web (Django, Flask entre autres) et autres gros projets (Ansible, Bootstrap, etc.). C'est une librairie très stable et très utilisée.

Du coup dans ma proposition, on n'utiliserait pas les codes POSIX de date (%y %m, etc.) mais vu la simplicité d'implémentation des templates et de leur rendu (deux lignes de code), ça me paraît justifié.

**Comment by PierreS-alpha** [comment]
Parfait, Jinja à l'air parfaitement indiqué pour le job.
Je vais regarder ses possibilités

**Reply by PierreS-alpha** [reply]
En fait il faut distinguer 2 aspects de la représentation du temps dans le fichier de config.

1. ce qui relève effectivement de l'aspect "template pattern":  Les champs du template seront remplacés automatiquement par les bonnes valeurs pour ce qui est du rangement des fichiers par exemple.
2. ce qui relève de la définition des intervalles pour les "actions" (i.e. download, convert, splicing etc...). Pour ça, c'est l'opérateur qui dit ce qu'il veut, soit en valeur absolue, soit en valeur relative ("now", "yesterday", "one hour ago"). Ici il n'est pas question de chemin dans l’arborescence, mais de localisation dans le temps. Il faut donc trouver une représentation a mi-chemin entre quelque chose de très verbeux (ce que j'ai mis ci avant serait une représentation poussée à l’extrême) et quelque chose 

Pour répondre à ta question @cyprien-griot, oui, il y aurait sans doute a décrire des opérations (simples) dans le fichier de config pour permettre cela

De plus, je parle de représentation du temps dans le fichier de config qui serait les taches routinières, mais dans mon idée on pourrait les outrepasser (override) avec des arguments en ligne de commande pour des re-téléchargements/re-conversions exceptionnels
